gapic/src/main/com/google/gapid/perfetto/models/SliceTrack.java

```diff
@@ -67,8 +67,48 @@ protected Slice buildSlice(Row row, ArgSet args) {
 
   public static SliceTrack forGpuQueue(QueryEngine qe, GpuInfo.Queue queue) {
     return new WithQueryEngine(qe, "gpu_slice", queue.trackId) {
+      private final String GPU_COLUMNS = "render_target, render_target_name, render_pass, render_pass_name, command_buffer, command_buffer_name";
+      private final String GPU_SLICE_SQL = "select " + BASE_COLUMNS + ", " + GPU_COLUMNS + " from %s where id = %d";
+      private final String GPU_SLICES_VIEW = "select " + BASE_COLUMNS + ", " + GPU_COLUMNS + " from %s where track_id = %d";
+
+      private String gpuSliceSql(long id) {
+        return format(GPU_SLICE_SQL, tableName("slices"), id);
+      }
+
+      @Override
+      protected ListenableFuture<?> initialize() {
+        String slices = tableName("slices");
+        String window = tableName("window");
+        String span = tableName("span");
+        return qe.queries(
+            dropTable(span),
+            dropView(slices),
+            dropTable(window),
+            createWindow(window),
+            createView(slices, format(GPU_SLICES_VIEW, table, trackId)),
+            createSpan(span, window + ", " + slices + " PARTITIONED depth"));
+      }
+
+      private boolean isParentSlice(ArgSet args) {
+        Object w = args.get("width");
+        if (w == null) {
+          return false;
+        }
+        return true;
+      }
+
+      @Override
+      public ListenableFuture<Slice> getSlice(long id) {
+        return transformAsync(expectOneRow(qe.query(gpuSliceSql(id))), r ->
+        transform(qe.getArgs(r.getLong(8)), args -> buildSlice(r, args)));
+      }
       @Override
       protected Slice buildSlice(Row row, ArgSet args) {
+        if (isParentSlice(args)) {
+          args.addEntry("VkFrameBuffer", row.getString(10) + "(0x" + row.getLong(9) + ")");
```

Comment:
Just noticed that these need to be converted to hex first.. Updating that now...